src/main/java/au/org/emii/ncdfgenerator/NcdfGenerator.java

```diff
@@ -0,0 +1,60 @@
+
+package au.org.emii.ncdfgenerator;
+
+import au.org.emii.ncdfgenerator.cql.ExprParser;
+import au.org.emii.ncdfgenerator.cql.IExprParser;
+import au.org.emii.ncdfgenerator.cql.IDialectTranslate;
+import au.org.emii.ncdfgenerator.cql.PGDialectTranslate;
+
+import java.sql.Connection;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.OutputStream;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+
+public class NcdfGenerator
+{
+	final IExprParser parser;
+	final IDialectTranslate translate;
+	final ICreateWritable createWritable;
+	final String layerConfigDir;
+
+	public NcdfGenerator( String layerConfigDir, String tmpCreationDir  )
```

Comment:
The path is dependent on the use context of this library. Eg. as a stand-alone library, or when embedded in geoserver, or when running an integration test. The library shouldn't be responsible for manage these details that depend on integration context. It's really waiting for spring configuration, and the ability to source these details from the container.